lua/telescope/actions/set.lua

```diff
@@ -40,6 +39,24 @@ set.select = function(prompt_bufnr, type)
   return set.edit(prompt_bufnr, action_state.select_key_to_edit_key(type))
 end
 
+local edit_buffer
+do
```

Comment:
Testing now but another tip :)
```lua
local edit_buffer = (function()
  local map = {}

  return function(command, bufnr)
    ...
  end
end)()
```

Runs first time (and only once) and initializes `edit_buffer`.

Reply:
Thats cool. I think this is the purpose of `do .. end` though and in my opinion `do .. end` is clearer. I think this is small so I hope you are okay with me just leaving it :smile:

Reply:
Do end just opens a block. Its a little bit different but i won't argue about some syntax sugar :laughing: Both works fine